echo_handlers.go

```diff
@@ -167,15 +167,13 @@ func (svc *Service) AppsShowHandler(c echo.Context) error {
 	appPermissions := []AppPermission{}
 	svc.db.Where("app_id = ?", app.ID).Find(&appPermissions)
 
-	requestMethods := []string{"pay_invoice"}
-	// because pay_invoice is enabled even
-	// when there are no permissions set
+	requestMethods := []string{}
 	for _, appPerm := range appPermissions {
 		if appPerm.RequestMethod == NIP_47_PAY_INVOICE_METHOD {
+			//find the pay_invoice-specific permissions
```

Comment:
What if there is no pay permission for this app? won't this break?

Reply:
Edit: I guess not. Could it be renamed to `appPayInvoicePermission`?

Reply:
Maybe the AppPermission is the wrong place for the expiry, since it should be the same across all permissions for an app?

Reply:
I don't see any problems with it. Yes it will be the same across all permissions.

Reply:
Could we get the first permission of the app if it exists then to display the expiry in `show.html`? right now it will not show the expiry for apps that don't have a pay permission

Reply:
If we really don't wish to give `payInvoice` special treatment then ig we shouldn't also send the `paySpecificPermission` to the frontend separately. Let's send `ExpiresAt`, `MaxAmount` and `BudgetRenewal` just like we send `BudgetUsage`

```js
renewsIn := ""
budgetRenewal := ""
budgetUsage := int64(0)
maxAmount := 0
expiresAt := time.Time{}

for _, appPerm := range appPermissions {
	if expiresAt.isZero() && !appPerm.ExpiresAt.isZero() {
		expiresAt = appPerm.ExpiresAt
	}
	if appPerm.RequestMethod == NIP_47_PAY_INVOICE_METHOD {
		//find the pay_invoice-specific permissions
		if appPerm.MaxAmount > 0 {
			budgetUsage = svc.GetBudgetUsage(&appPerm)
			budgetRenewal = appPerm.BudgetRenewal
			endOfBudget := GetEndOfBudget(budgetRenewal, app.CreatedAt)
			renewsIn = getEndOfBudgetString(endOfBudget)
		}
	}
	requestMethods = append(requestMethods, nip47MethodDescriptions[appPerm.RequestMethod])
}

// and then send all of them to frontend
```